R/openapi-types.R

```diff
@@ -0,0 +1,120 @@
+# calculate all OpenAPI type information at once and use created information throughout package
+dataTypesInfo <- list()
+dataTypesMap <- list()
+defaultDataType <- structure("string", default = TRUE)
+defaultSerialization <- structure(FALSE, default = TRUE)
+
+local({
+  addDataTypeInfo <- function(dataType, plumberTypes,
+                              regex = NULL, converter = NULL,
+                              regexSerialization = NULL,
+                              converterSerialization = NULL,
+                              format = NULL,
+                              location = NULL,
+                              realType = NULL) {
+    dataTypesInfo[[dataType]] <<-
+      list(
+        regex = regex, regexSerialization = regexSerialization,
+        converter = converter, converterSerialization = converterSerialization,
+        format = format,
+        location = location,
+        serializationSupport = !is.null(regexSerialization) & !is.null(converterSerialization),
+        realType = realType
+      )
+
+    for (plumberType in plumberTypes) {
+      dataTypesMap[[plumberType]] <<- dataType
+    }
+    # make sure it could be called again
+    dataTypesMap[[dataType]] <<- dataType
+
+    invisible(TRUE)
+  }
+
+  addDataTypeInfo(
+    "boolean",
+    c("bool", "boolean", "logical"),
+    "[01tfTF]|true|false|TRUE|FALSE",
+    as.logical,
+    "(?:(?:[01tfTF]|true|false|TRUE|FALSE),?)+",
+    function(x) {as.logical(stri_split_fixed(x, ",")[[1]])},
```

Comment:
Do not define the regex twice. 
Do not define this function split function.

Move the generation of these two objects into addDataTypeInfo and store them as different args. `regexArray`, `converterArray`